source/common/upstream/upstream_impl.h

```diff
@@ -490,6 +494,37 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
   uint64_t pending_initialize_health_checks_{};
 };
 
+/**
+ * Manage priority state of a cluster.
+ */
+class PriorityStateManager {
+public:
+  PriorityStateManager(ClusterImplBase& cluster, const LocalInfo::LocalInfo& local_info);
+
+  void
+  initializePerPriority(const envoy::api::v2::endpoint::LocalityLbEndpoints& locality_lb_endpoint);
+  void
+  registerHostPerPriority(const std::string& hostname,
+                          Network::Address::InstanceConstSharedPtr address,
+                          const envoy::api::v2::endpoint::LocalityLbEndpoints& locality_lb_endpoint,
+                          const envoy::api::v2::endpoint::LbEndpoint& lb_endpoint);
+  void registerHostPerPriority(const HostSharedPtr& host, const uint32_t priority);
+
+  void updateClusterPrioritySet(const uint32_t priority,
```

Comment:
Maybe add some comments explaining the contract/relationship between this and the above two methods.